### PR TITLE
Install sosreport controller plugin into proper path

### PR DESCRIPTION
**Pull request by anxstj**
##### SUMMARY

This MR fixes three things:
1) executing `controller.py` results in the following error:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sos/sosreport.py", line 1117, in setup
    plug.setup()
  File "/usr/local/lib/python3.9/site-packages/sos/plugins/controller.py", line 72, in setup
    self.add_forbidden_path(SOSREPORT_FORBIDDEN_PATHS)
  File "/usr/local/lib/python3.9/site-packages/sos/plugins/__init__.py", line 353, in add_forbidden_path
    for filespec in glob.glob(forbiddenPath):
  File "/usr/lib64/python3.9/glob.py", line 22, in glob
    return list(iglob(pathname, recursive=recursive))
  File "/usr/lib64/python3.9/glob.py", line 43, in _iglob
    dirname, basename = os.path.split(pathname)
  File "/usr/lib64/python3.9/posixpath.py", line 103, in split
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not list
```
The `add_forbidden_path()` function expects a path and not a list. So the script must iterate over the list and add each item separately.

2) Having a sosconfig path of `/usr/share/sosreport/sos/plugins` will install the script in `/var/lib/awx/venv/awx/lib/python3.9/site-packages/usr/share/sosreport/sos/plugins/controller.py`. But sosreport requires the script to be installed in the [sos.report.plugins](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#where-do-they-go) package path.

3) Install sosreport into the awx venv

So with these changes, it will be possible to execute `sosreport -o controller` in the AWX container.

##### ISSUE TYPE

 - Bug

##### COMPONENT NAME

 - API

##### AWX VERSION

```
20.1.1.dev41+g4dd6205054
```

### CODE REVIEW

**Comment by anxstj** [comment]
If this should work with the sos RPM package, then the plugin must be installed into `/usr/lib/python3.9/site-packages/sos/report/plugins`. That's probably not easily accomplishable with the `setup.py` installation method.

File: requirements/requirements.txt [hunk 1]
```diff
@@ -384,6 +384,7 @@ social-auth-core==4.2.0
     #   social-auth-app-django
 sqlparse==0.3.1
     # via django
+sosreport==3.2
```
Comment (AlanCoding) [inline requirements/requirements.txt]: I expect this is likely to conflict with other install methods. This can be a complicated balancing act.

Comment (anxstj) [reply requirements/requirements.txt]: Yes, that doesn't look right. It seems that CentOS/RHEL ships another version (4.x) that isn't available via pypi.

I will remove that piece and leave the sos utility installation to someone else.

**Comment by john-westcott-iv** [comment]
@anxstj Thank you so much for your pull request to AWX. This this PR was last looked at there are changes to the AWX code base that require this PR to be rebased. Would you be willing to perform a rebase to update this PR and then we can get it into our work stream?

**Comment by anxstj** [comment]
OK, I rebased it. It now contains only the small controller fix.

Adding it to the final image would need something like:
```diff
--- a/tools/ansible/roles/dockerfile/templates/Dockerfile.j2
+++ b/tools/ansible/roles/dockerfile/templates/Dockerfile.j2
@@ -111,6 +111,7 @@ RUN dnf -y update && dnf install -y 'dnf-command(config-manager)' && \
     rsync \
     "rsyslog >= 8.1911.0" \
     subversion \
+    sos \
     sudo \
     vim-minimal \
     which \
@@ -193,6 +194,8 @@ ADD tools/ansible/roles/dockerfile/files/wait-for-migrations /usr/local/bin/wait
 ADD tools/ansible/roles/dockerfile/files/stop-supervisor /usr/local/bin/stop-supervisor
 
 ## File mappings
+ADD tools/sosreport/controller.py /tmp/controller.py
+RUN chmod 644 /tmp/controller.py && mv /tmp/controller.py "$(find $(find /usr/lib -name sos) -name plugins)/controller.py"
 {% if build_dev|bool %}
 ADD tools/docker-compose/launch_awx.sh /usr/bin/launch_awx.sh
 ADD tools/docker-compose/nginx.conf /etc/nginx/nginx.conf
```
I'm not sure if you're willing to accept this.

**Review by jbradberry — APPROVED** [review]

**Comment by jbradberry** [comment]
@AlanCoding do we need to follow up with a separate ticket to update package installation?

**Comment by AlanCoding** [comment]
Maybe there's some additional unrelated work that we could do that for. But this looks fine to merge as far as I know.

**Comment by jay-steurer** [comment]
Hi @anxstj could you rebase your branch? 
Thanks mate

**Comment by anxstj** [comment]
@jay-steurer sure, done.

**Comment by jay-steurer** [comment]
Verified by QE. Merging it.

**Pull request merged**